auraed/src/runtime/cell_service/executables/auraed/isolation_controls.rs

```diff
@@ -0,0 +1,126 @@
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use libc::c_char;
+use std::io::{self};
+use std::path::PathBuf;
+use tracing::info;
+
+#[derive(Debug, Clone, Default)]
+pub struct IsolationControls {
+    pub isolate_process: bool,
+    pub isolate_network: bool,
+}
+
+#[derive(Default)]
+pub(crate) struct Isolation {
+    name: String,
+}
+
+impl Isolation {
+    pub fn new(name: &str) -> Isolation {
+        Isolation { name: name.to_string() }
+    }
+    pub fn setup(&mut self, iso_ctl: &IsolationControls) -> io::Result<()> {
+        // The only setup we will need to do is for isolate_process at this time.
+        // We can exit quickly if we are sharing the process controls with the host.
+        if !iso_ctl.isolate_process {
+            return Ok(());
+        }
+
+        // Bind mount root:root with MS_REC and MS_PRIVATE flags
+        // We are not sharing the mounts at this point (in other words we are in a new mount namespace)
+        let root = PathBuf::from("/");
+        nix::mount::mount(
+            Some(&root),
+            &root,
+            None::<&str>, // ignored
+            nix::mount::MsFlags::MS_PRIVATE | nix::mount::MsFlags::MS_REC,
+            None::<&str>, // ignored
+        )
+        .map_err(|e| io::Error::from_raw_os_error(e as i32))?;
+        info!("Isolation: Bind mounted root dir (/) in cell");
```

Comment:
I think strictly speaking this is not a bind mount as the BIND flag is not set.

Reply:
I’ll update — I also have an erroneous println in the logger as well I need to clean up